src/Adapter/Product/Combination/CommandHandler/SetCombinationSuppliersHandler.php

```diff
@@ -0,0 +1,108 @@
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Combination\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\Product\Combination\Repository\CombinationRepository;
+use PrestaShop\PrestaShop\Adapter\Product\Update\ProductSupplierUpdater;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Command\SetCombinationSuppliersCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\CommandHandler\SetCombinationSuppliersHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\ValueObject\CombinationId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Supplier\ProductSupplier as ProductSupplierDTO;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use ProductSupplier;
+
+final class SetCombinationSuppliersHandler implements SetCombinationSuppliersHandlerInterface
+{
+    /**
+     * @var CombinationRepository
+     */
+    private $combinationRepository;
+
+    /**
+     * @var ProductSupplierUpdater
+     */
+    private $productSupplierUpdater;
+
+    /**
+     * @param CombinationRepository $combinationRepository
+     * @param ProductSupplierUpdater $productSupplierUpdater
+     */
+    public function __construct(
+        CombinationRepository $combinationRepository,
+        ProductSupplierUpdater $productSupplierUpdater
+    ) {
+        $this->combinationRepository = $combinationRepository;
+        $this->productSupplierUpdater = $productSupplierUpdater;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(SetCombinationSuppliersCommand $command): array
+    {
+        $combination = $this->combinationRepository->get($command->getCombinationId());
+        $productId = new ProductId((int) $combination->id_product);
+
+        $productSuppliers = [];
+        foreach ($command->getCombinationSuppliers() as $productSupplierDTO) {
+            $productSuppliers[] = $this->buildEntityFromDTO($productId, $command->getCombinationId(), $productSupplierDTO);
+        }
+
+        return $this->productSupplierUpdater->setCombinationSuppliers(
+            $productId,
+            $command->getCombinationId(),
+            $productSuppliers
+        );
+    }
+
+    /**
+     * @param ProductId $productId
+     * @param CombinationId $combinationId
+     * @param ProductSupplierDTO $productSupplierDTO
+     *
+     * @return ProductSupplier
+     */
+    private function buildEntityFromDTO(
+        ProductId $productId,
+        CombinationId $combinationId,
+        ProductSupplierDTO $productSupplierDTO
+    ): ProductSupplier {
+        $id = $productSupplierDTO->getProductSupplierId() ? $productSupplierDTO->getProductSupplierId()->getValue() : null;
+
+        $productSupplier = new ProductSupplier();
```

Comment:
```suggestion
        $productSupplier = new ProductSupplier($id);
```

Creation and update shouldn't be handled the same way, when the ID is present we should use it to fetch the existing data, this brings two interest:
- we don't manage it yet but for partial updates we need to get existing object and update only required fields, here all fields are overwritten but some of them could be null in the future
- using the ID should allow to check that the ProductSupplier does exist in database, in that spirit maybe we should even rely on `ProductSupplierRepository` to get the object while checking its existence